executor/executor_test.go

```diff
@@ -8736,19 +8738,73 @@ func (s *testResourceTagSuite) TestResourceGroupTag(c *C) {
 	}
 
 	cases := []struct {
-		sql    string
-		ignore bool
+		sql       string
+		tagLabels []tipb.ResourceGroupTagLabel
+		ignore    bool
 	}{
-		{sql: "insert into t values(1,1),(2,2),(3,3)"},
-		{sql: "select * from t use index (idx) where a=1"},
-		{sql: "select * from t use index (idx) where a in (1,2,3)"},
-		{sql: "select * from t use index (idx) where a>1"},
-		{sql: "select * from t where b>1"},
-		{sql: "begin pessimistic", ignore: true},
-		{sql: "insert into t values(4,4)"},
-		{sql: "commit", ignore: true},
-		{sql: "update t set a=5,b=5 where a=5"},
-		{sql: "replace into t values(6,6)"},
+		{
+			sql: "insert into t values(1,1),(2,2),(3,3)",
+			tagLabels: []tipb.ResourceGroupTagLabel{
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelIndex,
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelIndex,
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelIndex,
+			},
+		},
+		{
+			sql: "select * from t use index (idx) where a=1",
+			tagLabels: []tipb.ResourceGroupTagLabel{
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelIndex,
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelRow,
+			},
+		},
+		{
+			sql: "select * from t use index (idx) where a in (1,2,3)",
+			tagLabels: []tipb.ResourceGroupTagLabel{
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelIndex,
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelRow,
+			},
+		},
+		{
+			sql: "select * from t use index (idx) where a>1",
+			tagLabels: []tipb.ResourceGroupTagLabel{
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelIndex,
+				tipb.ResourceGroupTagLabel_ResourceGroupTagLabelRow,
+			},
```

Comment:
You can add `select a from t use index (idx) where a>1`. We can expect pure index access for this case (as there is no index lookup)